bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/ImageProcessingDescriptor.java

```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.swt.internal.image;
+
+import org.eclipse.swt.graphics.*;
+
+public interface ImageProcessingDescriptor {
+
+	RGBA adaptPixelValue(RGBA original);
```

Comment:
At least with the current callers it would save us the intermediate object creation if this method would accept the arguments `int red, int green, int blue, int alpha` directly.

Reply:
Valid point. That reduces conversion efforts. I have adapted the PR accordingly.